controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -160,6 +160,12 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(ctx context.Context
 		log.Error(err, "Error initializing enactment")
 	}
 
+	enactmentInstance, err := r.applyEnactmentDefaults(instance)
```

Comment:
I would move this code to be inside `initializeEnactment` so it will be created with the default from the beginning and there won't be a need for 2 extra api calls (retrieving and updating the enactment).

Reply:
I'm not sure if moving this code into `initEnactment` is worth saving the calls, as it would make the function a little too complex IMO.
Mainly because `inititalizeEnactment` returns conditions of the 
WDYT about a compromise - saving 1 API call - by setting the desiredState with defaults in the `initializeEnactment` function and instead of this code only fetch the updated enactment?
That saves one API (write) call.

The main issue I have with moving this whole piece to `initializeEnactment` is that the function returns previous enactment  Status, and I don't like the idea of mixing the previous enactmentStatus with the current one.

Reply:
addressed in separate commit for easier review https://github.com/nmstate/kubernetes-nmstate/pull/793/commits/c1ec4a4fb04ba0f8c95825112b40c053a032b5c8